Feature/cheyenne gnu

natalie-perlin opened the pull request:
Changes for Cheyenne to build with modules restored/saved in module collection, to prevent over-loading system-installed modules with earlier versions

kgerheiser committed:
Fix BUILD_TESTS for bufr

Change TETS to TESTS

kgerheiser committed:
Merge pull request #372 from NOAA-EMC/feature/cheyenne

Fixes a few issues that were breaking the full build of stack_ufs_weather_ci.yaml on Cheyenne.

kgerheiser committed:
Merge branch 'develop' into bugfix/bufr-tests

Hang-Lei-NOAA committed:
part of issue #373"

kgerheiser committed:
Merge pull request #375 from Hang-Lei-NOAA/crtm_v2.4.0

Update CRTM to v2.4.0 which comes from a different remote.

aerorahul committed:
Add `csh` support for loading miniconda3 (#377)

kgerheiser committed:
Merge branch 'develop' into bugfix/bufr-tests

kgerheiser committed:
Merge pull request #368 from NOAA-EMC/bugfix/bufr-tests

Fix BUILD_TESTS for bufr

kgerheiser committed:
Skip build instead of exiting with failure if library exists (#364)

Skip build instead of exiting with failure if library exists

kgerheiser committed:
Re-name png to libpng (#389)

Make name consistent with project name and Spack install

kgerheiser committed:
Add stack_gfsv16.2.yaml and module variables (#392)

* Fix duplicate symbols in NCO

* Add stack_gfsv16.2.yaml

* Update modulefiles to support gfsv16.2

* Fix typo

* Update NCO version

* Fix CRTM fix file installation

* Use -qopenmp instead of -openmp

* Use --no-check-certificate on Orion because of OS cert issue

* Detect libdir dynamically for bufr using lua in the modulefile

* Move UPP build into its own build script and support makefile build

Commit:
Fix UPP build by setting myFC var

Commit:
Remve MPI build that was accidentally introduced

kgerheiser committed:
Merge pull request #397 from kgerheiser/bugfix/gfsv16-build

Bugfixes for GFSv16 build

natalie-perlin committed:
Add Intel CentOS7 config (#394)

* Adapted to linux-centos-intel

General Linux-CentOS (7), Intel compilers and MPI

* Update build_libtiff.sh

* Update build_proj.sh

kgerheiser committed:
Fix NCO build on Hera 

Disable ncap2 which requires Antlr libraries. This disables the C++ NCO libraries which we don't use.

kgerheiser committed:
Merge pull request #399 from NOAA-EMC/bugfix/nco-ncap2

Fix NCO build on Hera

Hang-Lei-NOAA committed:
update crtm #400

gspetro-NOAA committed:
HPC-Stack Documentation (#385)

Add Sphinx docs to hpc-stack

kgerheiser committed:
Merge branch 'develop' into feature/updatecrtm

Hang-Lei-NOAA committed:
Merge pull request #401 from Hang-Lei-NOAA/feature/updatecrtm

update crtm #400

kgerheiser committed:
Use ipolates=1 for wgrib2 (#404)

wgrib2 2.0.7 does not support ip=3

climbfuji committed:
Update Cheyenne configurations for Intel and GNU (#406)

Commit:
Changes for Cheyenne to build with modules restored/saved in module collection

Module collections are saved in /glade/u/home/nperlin/.lmod.d/ directory on Cheyenne

Commit:
updated build for Cheyenne

kgerheiser commented:
Are you merging into the correct branch? Do you want to merge into develop?

climbfuji commented on config/config_cheyenne_gnu.sh:
```diff
@@ -2,34 +2,34 @@
+export WGET="wget -nv --no-check-certificate"
 
 module purge
-module unuse /glade/u/apps/ch/modulefiles/default/compilers
```
Have you tried this with netcdf 4.8.1? My guess is that you will run into trouble, but hopefully I am wrong.

natalie-perlin replied:
Yes, this build was made with netcdf version_c 4.8.1 and version_f 4.5.4

natalie-perlin committed:
Update build_esmf.sh

Downloading of the software for a specific tag of the ESMF (not a branch), need to use different git syntax:
git  clone $URL $software

natalie-perlin committed:
Update stack_cheyenne.yaml

Removed flags for STACK_w3nco_FFLAGS, STACK_w3nco_FCFLAGS

natalie-perlin commented:
@kgerheiser - a branch feature/mac probably needs merging with develop first! And then feature/cheyenne could be checked on merging conflict. This may or may not needs to be merged with the develop branch, as it extensively uses functionality of Lmod to save/restore module collections.

natalie-perlin commented:
Will close this branch, and further do pull requests into existing feature/cheyenne branch
